mlir/lib/Dialect/Bufferization/Transforms/OwnershipBasedBufferDeallocation.cpp

```diff
@@ -48,6 +48,32 @@ static Value buildBoolValue(OpBuilder &builder, Location loc, bool value) {
 
 static bool isMemref(Value v) { return v.getType().isa<BaseMemRefType>(); }
 
+/// Return "true" if the given op is guaranteed to have no "Allocate" or "Free"
```

Comment:
This is super nit picky but the not before an or is always ambiguous (as in !Allocate or Free or !(Allocate or Free). I think you can avoid this by just saying to have neither Allocate nor Free ....

(and yes, I realize this is very nitpicky)